sdk/helper/ldaputil/ldap.go

```diff
@@ -13,16 +15,19 @@ func NewLDAP() LDAP {
 // LDAP provides ldap functionality, but through an interface
 // rather than statically. This allows faking it for tests.
 type LDAP interface {
-	Dial(network, addr string) (Connection, error)
-	DialTLS(network, addr string, config *tls.Config) (Connection, error)
+	Dial(addr string) (Connection, error)
+	DialTLS(addr string, config *tls.Config) (Connection, error)
 }
 
 type ldapIfc struct{}
 
-func (l *ldapIfc) Dial(network, addr string) (Connection, error) {
-	return ldap.Dial(network, addr)
+func (l *ldapIfc) Dial(addr string) (Connection, error) {
+	dialer := ldap.DialWithDialer(&net.Dialer{Timeout: 10 * time.Second})
```

Comment:
nit: Can we make this timeout a constant? It might also make a good configurable.